### PR TITLE
Standardize node version manager - Closes #1204

### LINKED ISSUE #1204: Standardize node version manager recommendation to nvm
I.e. in readme. Should also cover e.g. a .nvmrc file and `engines` section of `package.json`.

This issue is about standardizing the recommendation to nvm in line with our recommendations and usage in other projects, in support of a timely release of `1.0.0`. https://github.com/LiskHQ/lisk-template/issues/70 is an issue to revisit this choice on an organization-wide basis, so any preferences regarding version managers should be logged there.

Child of #672

### PR DESCRIPTION
**Pull request by ManuGowda**
Closes #1204

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,12 +67,24 @@ This sections provides details on what you need install on your system in order
...
+    - Linux
+
+        ```
+        curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
+        nvm install 6.12.3
+        ```
+
+    - Mac
+        ```
+        brew update
```
Comment (willclarktech) [inline README.md]: We should be consistent about whether we explicitly recommend updating brew.

File: README.md [hunk 2]
```diff
@@ -67,12 +67,24 @@ This sections provides details on what you need install on your system in order
         brew install node@6
```
Comment (willclarktech) [inline README.md]: This should be updated to our specific node version.

File: README.md [hunk 3]
```diff
@@ -67,12 +67,24 @@ This sections provides details on what you need install on your system in order
...
+        nvm install 6.12.3
+        ```
+
+    - Mac
+        ```
+        brew update
+        brew install nvm
+        mkdir ~/.nvm
+        echo "export NVM_DIR=~/.nvm" >> ~/.bash_profile
+        echo "source $(brew --prefix nvm)/nvm.sh" >> ~/.bash_profile
```
Comment (willclarktech) [inline README.md]: These only apply to bash users.

File: README.md [hunk 4]
```diff
@@ -66,27 +66,14 @@ This sections provides details on what you need install on your system in order
     - MacOS 10.12-10.13 (Sierra/High Sierra):
 
         ```
-        brew install node@6
+        brew install node@6.12.3
         ```
 
-- _(Recommended)_ Install nvm -- Used for Node.js version management
+- _(Recommended)_ Install nvm(<https://github.com/creationix/nvm#install-script>) -- Used for Node.js version management
```
Comment (willclarktech) [inline README.md]: Maybe this makes more sense under the previous bullet point so people don't install Node.js system-wide and then see that we recommend not doing that?

```markdown
_(Recommended)_ Install using a version manager such as nvm.
    1. Install nvm following their instructions (https://github.com/creationix/nvm#installation)
    1. Install the correct version of Node.js using nvm:
    ```
    nvm install 6.12.3
    ```

Comment (ManuGowda) [reply README.md]: yeah I think keeping the nvm install under bullet point would enforce them install nvm and then switch to specific node version.

**Pull request merged**